src/CMakeLists.txt

```diff
@@ -157,7 +157,8 @@ endif ()
 add_library(dbcsr ${DBCSR_SRCS})
 
 set_target_properties(dbcsr PROPERTIES VERSION ${dbcsr_VERSION}
-                                       SOVERSION ${dbcsr_APIVERSION})
+                                       SOVERSION ${dbcsr_APIVERSION}
+			               POSITION_INDEPENDENT_CODE ${BUILD_SHARED_LIBS})
```

Comment:
This is only required if `dbcsr` were a `STATIC` library, which it isn't.

Reply:
It does not matter actually. Static libraries can also be build with -fPIC (usually even suggested when we mix static and dynamic libraries). I changed it `POSITION_INDEPENDENT_CODE ON` all the time.

Reply:
> It does not matter actually. Static libraries can also be build with -fPIC (usually even suggested when we mix static and dynamic libraries). I changed it `POSITION_INDEPENDENT_CODE ON` all the time.

Indeed, very advisable to use Position Independent Code even for all objects even when collected into an archive aka static library. Nowadays, the reason things work without `-fPIC` is that some toolchains apply it by default, e.g., one-step-builds (compile and link into one command). Using `-fPIC` everywhere also technically allows to use the same objects twice (for archive and shared library).